### PR TITLE
Scripts/Silverpine Forest: Fix Silverpine Forest till quest 27601

### PR DESCRIPTION
**Pull request by Nyr97**
**Changes proposed**:

There are several quests within the entire questline from Silverpine Forest that need retouching yet due to a few inaccuracies due to lack of sniffs. They will be deployed within the next days. DB fixes will be out soon for PR.

Please, feel free to advise or improve the current state of the zone in any possible way. Also, if you have any data regarding Silverpine Forest, make sure to share them with me so I can either implement or compare at any point.

**Issues addressed**: Silverpine Forest questline

**Tests performed**: It builds and it was tested in game.

**Known issues and TODO list**:

- [ ] Quest 27045 "Waiting to Exsanguinate" not yet ready.
- [ ] Quest 27099 "No Escape" needs some fixes.
- [ ] Quest 27098 "Lordaeron" needs some retouching on Sylvanas' and player's pathing.
- [ ] Quest 27232 "The Waters Run Red" not yet ready.
- [ ] Quest 27195 "Nowhere to Run" not yet ready.
- [ ] Quest 27601 "Cities in Dust" not yet ready.

- [ ] DB fixes not yet deployed.

### CODE REVIEW

File: src/server/game/Spells/SpellMgr.cpp [hunk 1]
```diff
@@ -4870,7 +4870,119 @@ void SpellMgr::LoadSpellInfoCorrections()
...
+
+    ApplySpellFix({
+    83464,  // Chain Visual
+    83467,
+    86803,
+    86805,
+    84112   // Camera spell
+        }, [](SpellInfo* spellInfo)
+    {
+        spellInfo->Effects[EFFECT_0].TargetA = SpellImplicitTargetInfo(TARGET_UNIT_TARGET_ANY);
```
Comment (Ovahlord) [inline src/server/game/Spells/SpellMgr.cpp:4883]: hack. use conditions.

File: src/server/game/Spells/SpellMgr.cpp [hunk 2]
```diff
@@ -4870,7 +4870,119 @@ void SpellMgr::LoadSpellInfoCorrections()
...
+        }, [](SpellInfo* spellInfo)
+    {
+        spellInfo->Effects[EFFECT_0].TargetA = SpellImplicitTargetInfo(TARGET_DEST_DB);
+    });
+
+    ApplySpellFix({
+        83150, // Forsaken Trooper (Male 01)
+        }, [](SpellInfo* spellInfo)
+    {
+        spellInfo->Effects[EFFECT_0].MiscValue = 44596;
```
Comment (Ovahlord) [inline src/server/game/Spells/SpellMgr.cpp:4923]: why did you change the values of all these transform spells?

Comment (Nyr97) [reply src/server/game/Spells/SpellMgr.cpp:4923]: Apparently, there's a master spell that controls all of these individual spells. However, if left unchanged, the spell chooses a random creature_template ID to modify the creature's appearance, which would lead to inaccuracies, taking into account each individual creature transforms into a set creature. 

So each individual spell was supposed to be set to their connecting creature_template ID.

File: src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp [hunk 3]
```diff
@@ -68,7 +68,7 @@ class playerScript_silverpine_gilneas : public PlayerScript
     {
         if (newArea == 233) // Ambermill
         {
-            if (!player->GetQuestStatus(27518) == QUEST_STATUS_COMPLETE)
+            if (player->GetQuestStatus(27518) != QUEST_STATUS_COMPLETE)
                 player->CastSpell(player, 85361, true);
```
Comment (Ovahlord) [inline src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp:72]: no magic numbers here. use enums.

Comment (Nyr97) [reply src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp:72]: Sure, I'll fix it soon!

File: src/server/game/Spells/SpellMgr.cpp (hunk 2 above)
Comment (Nyr97) [reply src/server/game/Spells/SpellMgr.cpp:4923]: I have removed them, I'll push changes soon. I've just corrected the DB values.

**Comment by Nyr97** [comment]
I will be closing this PR in order to make smaller batches so that the review is easier. Especially for DB updates.
